components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -136,10 +141,29 @@ public class OpenIDConnectAuthenticator extends AbstractApplicationAuthenticator
     private static final String DYNAMIC_PARAMETER_LOOKUP_REGEX = "\\$\\{(\\w+)\\}";
     private static final String IS_API_BASED = "IS_API_BASED";
     private static final String REDIRECT_URL = "REDIRECT_URL";
+    private static final String SPACE_REGEX = "\\s+";
+    private static final String SPACE = " ";
+    private static final String SEMI_COLON_DELIMITER = ";";
+    private static final String COMMA_DELIMITER = ",";
     private static Pattern pattern = Pattern.compile(DYNAMIC_PARAMETER_LOOKUP_REGEX);
     private static final String[] NON_USER_ATTRIBUTES = new String[]{"at_hash", "iss", "iat", "exp", "aud", "azp"};
     private static final String AUTHENTICATOR_MESSAGE = "authenticatorMessage";
 
+    /**
+     * This method returns the current federated authenticator name. If there is no external IdP, then the current
+     * authenticator name is returned.
+     *
+     * @param context Authentication context.
+     * @return Federated authenticator name.
+     */
+    private String getFederatedAuthenticatorName(AuthenticationContext context) {
+
+        if (context == null || context.getExternalIdP() == null) {
```

Comment:
```suggestion
        if (context == null || context.getExternalIdP() == null || context.getExternalIdP().getIdPName() == null) {
```
this will make sure that a null value is not returned from this method.

Reply:
Fixed d2b40f9255b4a5ef3b8f9d44bd58ccdba9f50f70